client/extensions/woocommerce/lib/generate-variations/index.js

```diff
@@ -0,0 +1,36 @@
+// http://stackoverflow.com/questions/12303989/cartesian-product-of-multiple-arrays-in-javascript
+const f = ( a, b ) => [].concat( ...a.map( c => b.map( d => [].concat( c, d ) ) ) );
+const cartesian = ( a, b, ...c ) => b ? cartesian( f( a, b ), ...c ) : a;
+
+/**
+ * Generates variation objects based on a product's attributes.
+ *
+ * @param {Object} product Product object.
+ * @return {Array} Array of variation objects.
+ */
+export default function generateVariations( product ) {
+	const { attributes } = product;
```

Comment:
You could write `export default function generateVariations( { attributes } ) {` instead, I personally find that syntax one of the best shortcuts of ES6 :)